pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -680,6 +686,102 @@ func (c *VolumeZoneChecker) predicate(pod *v1.Pod, meta algorithm.PredicateMetad
 	return true, nil, nil
 }
 
+// ReplicasSpreadChecker contains the maximum replicas number check for a predicate.
+type ReplicasSpreadChecker struct {
+}
+
+func NewReplicasSpreadPredicate() algorithm.FitPredicate {
+	c := &ReplicasSpreadChecker{}
+	return c.predicate
+}
+
+func (c *ReplicasSpreadChecker) predicate(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	// check the limited number of replicas
+	if pod.Annotations != nil {
+		if maxReplicasNumber, ok := pod.Annotations[MaxReplicasPerHost]; ok {
+			maxReplicasNumberValue, err := strconv.Atoi(maxReplicasNumber)
+			if err != nil {
+				klog.Errorf("failed to parse MaxReplicasPerHost [%s] to a number: %v", maxReplicasNumber, err)
+			}
+			podsOnNode := nodeInfo.Pods()
```

Comment:
should we also support zone?

Reply:
Good point, thanks. although I have not consider it here, but it is worthing considering.